### PR TITLE
Changed topics type and refactored token management outcome parser

### PR DESCRIPTION
**Pull request by popenta**
Updated to match the sdk-specs.

### CODE REVIEW

File: multiversx_sdk/core/transactions_outcome_parsers/resources.py [hunk 1]
```diff
@@ -38,4 +40,42 @@ def __init__(self,
...
+                 return_data_parts: List[bytes] = [],
+                 return_message: str = "",
+                 return_code: str = "") -> None:
+        self.function = function
+        self.return_data_parts = return_data_parts
+        self.return_message = return_message
+        self.return_code = return_code
+
+
+def find_events_by_identifier(transaction_outcome: TransactionOutcome, identifier: str) -> List[TransactionEvent]:
```
[the remark this answers is not in the document]
Comment (popenta) [reply multiversx_sdk/core/transactions_outcome_parsers/resources.py:58]: will add it in the next PR.

File: multiversx_sdk/core/transactions_outcome_parsers/resources.py [hunk 2]
```diff
@@ -38,4 +40,42 @@ def __init__(self,
...
+) -> List[TransactionEvent]:
+    events = gather_all_events(transaction_outcome)
+    return list(filter(predicate, events))
+
+
+def gather_all_events(transaction_outcome: TransactionOutcome) -> List[TransactionEvent]:
+    all_events = [*transaction_outcome.logs.events]
+
+    for result in transaction_outcome.transaction_results:
+        all_events.extend([*result.logs.events])
```
[the remark this answers is not in the document]
Comment (popenta) [reply multiversx_sdk/core/transactions_outcome_parsers/resources.py:79]: Indeed. Changed.

File: multiversx_sdk/core/transactions_outcome_parsers/resources.py [hunk 3]
```diff
@@ -38,4 +40,42 @@ def __init__(self,
...
+        self.function = function
+        self.return_data_parts = return_data_parts
+        self.return_message = return_message
+        self.return_code = return_code
+
+
+def find_events_by_identifier(transaction_outcome: TransactionOutcome, identifier: str) -> List[TransactionEvent]:
+    events = find_events_by_predicate(transaction_outcome, lambda event: event.identifier == identifier)
+
+    if len(events) == 0:
```
[the remark this answers is not in the document]
Comment (popenta) [reply multiversx_sdk/core/transactions_outcome_parsers/resources.py:61]: Indeed. removed here as well.

File: multiversx_sdk/core/transactions_outcome_parsers/token_management_transactions_outcome_parser.py [hunk 4]
```diff
@@ -19,242 +18,233 @@ class TokenManagementTransactionsOutcomeParser:
...
 
-    def parse_register_meta_esdt(self, transaction_outcome: TransactionOutcome) -> RegisterMetaEsdtOutcome:
-        self._ensure_no_error(transaction_outcome.transaction_logs.events)
+        register_events = find_events_by_identifier(transaction_outcome, "registerAndSetAllRoles")
+        set_role_events = find_events_by_identifier(transaction_outcome, "ESDTSetRole")
 
-        event = self._find_single_event_by_identifier(transaction_outcome, "registerMetaESDT")
-        identifier = self._extract_token_identifier(event)
+        if len(register_events) != len(set_role_events):
+            raise ParseTransactionOutcomeError("Register Events and Set Role events mismatch. Should have the same number of events.")
```
[the remark this answers is not in the document]
Comment (popenta) [reply multiversx_sdk/core/transactions_outcome_parsers/token_management_transactions_outcome_parser.py:52]: Changed to:
```
The number of `registerAndSetAllRoles` events and `ESDTSetRole` events do not match`.
```

File: multiversx_sdk/core/transactions_outcome_parsers/resources.py [hunk 5]
```diff
@@ -38,4 +38,37 @@ def __init__(self,
                  transaction_results: List[SmartContractResult],
                  transaction_logs: TransactionLogs) -> None:
         self.transaction_results = transaction_results
```
Comment (ssd04) [inline multiversx_sdk/core/transactions_outcome_parsers/resources.py:40]: in specs, there is a reference to `direct_smart_contract_call_outcome: SmartContractCallOutcome;` should this be set here also?

Comment (popenta) [reply multiversx_sdk/core/transactions_outcome_parsers/resources.py:40]: That was added after this implementation was done. It is not needed now, will be added in a future PR.

**Review by andreibancioiu — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Pull request merged**